### PR TITLE
Fleet

### PR DESCRIPTION
**Pull request by lippfi**
One more set of changes for Fleet

### CODE REVIEW

File: src/main/java/com/maddyhome/idea/vim/ui/ex/ExEntryPanelService.kt [hunk 1]
```diff
@@ -95,6 +95,18 @@ public class ExEntryPanelService : VimCommandLineService {
     return panel
   }
 
+  public override fun createWithoutShortcuts(
```
Comment (AlexPl292) [inline src/main/java/com/maddyhome/idea/vim/ui/ex/ExEntryPanelService.kt]: `createSomething` usually used when we actually create a new instance of the object.
Here we reuse initialized instance of the ExEntryPanel, so I'd say that this method should be named also `getInstance....`

File: vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimCommandLine.kt [hunk 2]
```diff
@@ -11,6 +11,8 @@ package com.maddyhome.idea.vim.api
 import javax.swing.KeyStroke
 
 public interface VimCommandLine {
+  public val caret: VimCommandLineCaret
```
Comment (AlexPl292) [inline vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimCommandLine.kt:15]: I don't understand why this change is included in this commit.

File: src/main/java/com/maddyhome/idea/vim/ui/ex/ExTextField.java [hunk 3]
```diff
@@ -209,11 +207,11 @@ private void setFontToJField(String stringToDisplay) {
 
   void setEditor(@NotNull Editor editor, DataContext context) {
     this.context = context;
-    EditorHolderService.getInstance().setEditor(editor);
+    ExEntryPanel.getInstance().setEditor(editor);
   }
 
   public Editor getEditor() {
```
Comment (AlexPl292) [inline src/main/java/com/maddyhome/idea/vim/ui/ex/ExTextField.java:213]: This requires `@Nullable` on return type now

File: vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimSearchGroupBase.kt [hunk 4]
```diff
@@ -48,23 +48,35 @@ public abstract class VimSearchGroupBase : VimSearchGroup {
...
+      @JvmStatic
+      get
+      @JvmStatic
+      set
 
     /**
      * The type of the last used pattern.
      */
-    private var lastPatternType: PatternType? = null
+    @JvmStatic
```
Comment (AlexPl292) [inline vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimSearchGroupBase.kt:66]: Why do we need all new `@JvmStatic`? I see that their usage was moved from java class to kotlin class.

**Review by AlexPl292 — CHANGES_REQUESTED** [review]

File: src/main/java/com/maddyhome/idea/vim/ui/ex/ExEntryPanelService.kt (hunk 1 above)
Comment (lippfi) [reply src/main/java/com/maddyhome/idea/vim/ui/ex/ExEntryPanelService.kt]: In IdeaVim, we reuse the initialized instance, but this is unlikely to be the case for Fleet. The implementation details of IdeaVim should not influence naming of the new API
Previously, we used `getInstanceWithoutShortcuts()` and then called `activate()` on it. Now, however, we have adopted a more intuitive approach by directly calling `createWithoutShortcuts()`

File: vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimCommandLine.kt (hunk 2 above)
Comment (lippfi) [reply vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimCommandLine.kt:15]: I don't have any objective reason for this; I just felt that this field would look better at the top of the class and didn't want to create a separate commit for such a minor change
